lib/pbench/test/functional/server/test_datasets.py

```diff
@@ -565,15 +603,14 @@ def test_contents(self, server_client: PbenchServerClient, login_user):
             # directory is technically possible, but not legal unless it's a
             # trivial "archiveonly" dataset. NOTE: this will also fail if
             # either the "directories" or "files" JSON keys are missing.
-            archive = dataset.metadata["server.archiveonly"]
-            assert json["directories"] or json["files"] or archive
+            assert json["directories"] or json["files"]
```

Comment:
You don't want to test what does happen if someone queries an `archiveonly` dataset?

Reply:
I'm not sure what you mean: we know that we upload an archive only dataset, so we'll hit it here, and we expect it to take the `CONFLICT` path at line 594 ... where we're returning without checking the others...

Reply:
So, if someone queries an `archiveonly` dataset, they are supposed to get a `CONFLICT` and no JSON payload (or, there are no guarantees about what the payload might contain)?  Fair enough.